cmd/fpga_plugin/README.md

```diff
@@ -1,120 +1,269 @@
+    * [Choose the FPGA mode](#choose-the-fpga-mode)
+    * [Deploying as a DaemonSet](#deploying-as-a-daemonset)
+        * [Build the plugin image](#build-the-plugin-image)
+        * [Create a service account](#create-a-service-account)
+        * [Deploy the DaemonSet](#deploy-the-daemonset)
+        * [Verify plugin registration](#verify-plugin-registration)
+    * [Deploy by hand](#deploy-by-hand)
+        * [Build FPGA device plugin](#build-fpga-device-plugin)
+        * [Run FPGA device plugin in af mode](#run-fpga-device-plugin-in-af-mode)
+        * [Run FPGA device plugin in region mode](#run-fpga-device-plugin-in-region-mode)
+* [Next steps](#next-steps)
+
+# Introduction
+
+This FPGA device plugin is part of a collection of Kubernetes components found within this
+repository that enable integration of Intel FPGA hardware into Kubernetes.
+
+The following hardware platforms are supported:
+
+- Intel Arria 10
+- Intel Stratix 10
+
+The components support the [Open Programmable Acceleration Engine (OPAE)](https://opae.github.io/latest/index.html)
+and OpenCL application interfaces.
+
+The components together implement the following features:
+
+- discovery of pre-programmed accelerator functions
+- orchestration of FPGA programming
+- access control for FPGA hardware
+
+# Component overview
+
+The following components are part of this repository, and work together to support Intel FPGAs under
+Kubernetes:
+
+-   [FPGA device plugin](README.md) (this component)
+
+    A Kubernetes [device plugin](https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/)
+    that discovers available FPGA resources on a node and advertises them to the Kubernetes control plane
+    via the node kubelet.
 
--   [FPGA device plugin](README.md) (this module)
 -   [FPGA admission controller webhook](../fpga_admissionwebhook/README.md)
+
+    A Kubernetes [admission controller webhook](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/)
+    which can be used to dynamically convert logical resource names in pod specifications into actual FPGA
+    resource names, as advertised by the device plugin.
+
+    The webhook can also set environment variables to instruct the FPGA prestart hook to program the FPGA
+    before launching the container.
+
 -   [FPGA prestart CRI-O hook](../fpga_crihook/README.md)
 
+    A [CRI-O](https://github.com/cri-o/cri-o) prestart hook that, upon instruction from the FPGA admission
+    controller, allocates and programs the FPGA before the container is launched.
 
-### Get source code:
-```
-    $ mkdir -p $GOPATH/src/github.com/intel/
-    $ cd $GOPATH/src/github.com/intel/
-    $ git clone https://github.com/intel/intel-device-plugins-for-kubernetes.git
-```
+The repository also contains an [FPGA helper tool](../fpga_tool/README.md) that may be useful during
+development, initial deployment and debugging.
 
```

Comment:
fpgatool is also used by prestart hook to program devices.

Reply:
@bart0sh does it? I can't find that code - but reading the hook code, it looks like it is using the fpga libraries from the pkg subdir?

Reply:
True. They're both(the hook and the tool) use the same libraries. Sorry for the incorrect info.